specification/api-correlationcontext.md

```diff
@@ -149,37 +152,83 @@ Enter Scope 1
 Close Scope 1
 ```
 
-## DistributedContext
+## CorrelationContext
 
-### GetEntries
+### GetCorrelations
 
-Returns the entries in this `DistributedContext`.
-The order of entries is not significant. Based on the language specification,
-the returned value can be either an immutable collection or an immutable iterator
-to the collection of entries in this `DistributedContext`.
+Returns the entries in this `CorrelationContext`. The order of entries is not
+significant. Based on the language specification, the returned value can be
+either an immutable collection or an immutable iterator to the collection of
+entries in this `CorrelationContext`.
 
-### GetEntryValue
+### GetCorrelation
 
-Returns the `EntryValue` associated with the given `EntryKey`, or null if the given `EntryKey`
-is not present.
+To access the value for a label set by a prior event, the Correlations API
+provides a function which takes a context and a key as input, and returns a
+value. Returns the `EntryValue` associated with the given `EntryKey`, or null
+if the given `EntryKey` is not present.
 
 Required parameter:
 
+`Context` the context from which to get the correlation.
+
 `EntryKey` entry key to return the value for.
 
+### SetCorrelation
+
+To record the value for a label, the Correlations API provides a function which
+takes a context, a key, and a value as input. Returns an updated `Context` which
+contains the new value.
+
+Required parameter:
+
+`Context` the context to which the value will be set.
+
+`EntryKey` entry key for which to set the value.
+
+`EntryValue` entry value to set.
+
+### RemoveCorrelation
+
+To delete a label, the Correlations API provides a function which takes a context
+and a key as input. Returns an updated `Context` which no longer contains the selected `EntryKey`.
+
+Required parameter:
+
+`Context` the context from which to remove the correlation.
+
+`EntryKey` entry key to remove.
+
+### ClearCorrelations
+
+To avoid sending any labels to an untrusted process, the Correlation API provides
+a function to remove all Correlations from a context. Returns an updated `Context`
+with no correlations.
+
+Required parameter:
+
+`Context` the context from which to remove all correlations.
+
+### GetCorrelationPropagator
```

Comment:
I think we can remove this from the API spec. The Propagators will include a W3C correlation context propagator or some alternate propagator for correlation context. It's not an API for users.

Reply:
Removed